iceoryx_hoofs/test/moduletests/test_cxx_expected.cpp

```diff
@@ -334,22 +355,50 @@ TEST_F(expected_test, ErrorTypeOnlyCreateErrorLeadsToError)
     ASSERT_THAT(sut.get_error(), Eq(TestError::ERROR2));
 }
 
-TEST_F(expected_test, ErrorTypeOnlyMoveCtorLeadsToInvalidState)
+TEST_F(expected_test, ErrorTypeOnlyCreateValueWithoutValueMoveCtorLeadsToNoError)
 {
-    auto sut = expected<TestError>::create_error(TestError::ERROR2);
-    auto movedValue{std::move(sut)};
-    IOX_DISCARD_RESULT(movedValue);
-    ASSERT_TRUE(sut.has_error());
-    EXPECT_THAT(sut.get_error(), Eq(TestError::INVALID_STATE));
+    auto sutSource = expected<NonTrivialTestClass>::create_value();
+    auto sutDestination{std::move(sutSource)};
+    EXPECT_FALSE(sutSource.has_error());
+    EXPECT_FALSE(sutDestination.has_error());
+}
+
+TEST_F(expected_test, ErrorTypeOnlyCreateValueWithoutValueMoveAssignmentLeadsToNoError)
+{
+    auto sutSource = expected<NonTrivialTestClass>::create_value();
+    auto sutDestination = std::move(sutSource);
+    EXPECT_FALSE(sutSource.has_error());
```

Comment:
Same here, please add a check with verifies that `sutSource` was moved.

Reply:
Same here.